### PR TITLE
Remove endpoint name

### PR DESCRIPTION
**Pull request by bording**
This brings in the relevant parts from #3767 to remove the EndpointName class.

Connects to Particular/PlatformDevelopment#814.

### CODE REVIEW

File: src/NServiceBus.Core/Serialization/SerializationFeature.cs [hunk 1]
```diff
@@ -82,7 +82,7 @@ static void LogFoundMessages(IReadOnlyCollection<MessageMetadata> messageDefinit
                 return;
             }
             Logger.DebugFormat("Message definitions: \n {0}",
-                string.Concat(messageDefinitions.Select(md => md.ToString() + "\n")));
+                string.Concat(messageDefinitions.Select(md => md + "\n")));
```
Comment (timbussmann) [inline src/NServiceBus.Core/Serialization/SerializationFeature.cs]: nitpicking: should we use `string.Join` here?

**Comment by timbussmann** [comment]
no obsoletes required?

**Comment by andreasohlund** [comment]
Pretty sure this was all added in v6 so no need

**Comment by timbussmann** [comment]
> Pretty sure this was all added in v6 so no need

:+1: :shipit:

**Pull request merged**